lib/backend/k8s/resources/profile.go

```diff
@@ -229,10 +230,21 @@ func (c *profileClient) Watch(ctx context.Context, list model.ListInterface, rev
 	if !ok {
 		return nil, fmt.Errorf("ListInterface is not a ResourceListOptions: %s", list)
 	}
+	namePrefix := ""
 	if len(rlo.Name) != 0 {
 		// We've been asked to watch a specific profile.
 		log.WithField("name", rlo.Name).Debug("Watching a single profile")
-		opts.FieldSelector = fields.OneTermEqualSelector("metadata.name", rlo.Name).String()
+		if strings.HasPrefix(rlo.Name, "kns.") {
+			namePrefix = "kns"
+			shortName := strings.TrimPrefix(rlo.Name, "kns.")
```

Comment:
Let's use the existing conversion functions I linked you for these so we're note duplicating work.

Reply:
good stuff!